homeassistant/components/flo/device.py

```diff
@@ -0,0 +1,174 @@
+"""Flo device object."""
+from datetime import datetime, timedelta
+import logging
+from typing import Any, Callable, Dict, List
+
+from aioflo.api import API
+
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import SIGNAL_DEVICE_UPDATED, SIGNAL_WATER_CONSUMPTION_UPDATED
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class FloDevice:
+    """Flo device object."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        device_information: Dict[str, Any],
+        api_client: API,
+    ):
+        """Initialize the device."""
+        self.hass: HomeAssistantType = hass
+        self.api_client: API = api_client
+        self.unsubs: List[Callable] = []
+        self._flo_device_id: str = device_information["id"]
+        self._flo_location_id: str = device_information["location"]["id"]
+        self._mac_address: str = device_information["macAddress"]
+        self._manufacturer: str = "Flo by Moen"
+        self._model: str = device_information["deviceModel"]
+        self._device_type: str = device_information["deviceType"]
+        self._serial_number: str = device_information["serialNumber"]
+
+        self._available: bool = device_information["isConnected"]
+        self._firmware_version: str = device_information["fwVersion"]
+        self._rssi: int = device_information["connectivity"]["rssi"]
+        self._last_heard_from_time: str = device_information["lastHeardFromTime"]
+        self._current_system_mode: str = device_information["systemMode"]["lastKnown"]
+        self._target_system_mode: str = device_information["systemMode"]["target"]
+        self._current_flow_rate: float = device_information["telemetry"]["current"][
+            "gpm"
+        ]
+        self._current_psi: float = device_information["telemetry"]["current"]["psi"]
+        self._temperature: float = device_information["telemetry"]["current"]["tempF"]
+
+        self._consumption_today: float = None
+
+        self.unsubs.append(
```

Comment:
We shouldn't put side effects in constructors. 

Also, if all entities for this device are disabled, we are still fetching the data. Hence why the DataUpdateCoordinator is preferred.

Reply:
removed in favor of DataUpdateCoordinator